proposals/0010-resource-element-type-validation.md

```diff
@@ -0,0 +1,136 @@
+* Proposal: [0010](0010-resource-element-type-validation.md)
+* Author(s): [Joshua Batista](https://github.com/bob80905)
+* Sponsor: Joshua Batista
+* Status: **Under Consideration**
+* Impacted Project(s): (LLVM)
+
+* Issues: [#75676](https://github.com/llvm/llvm-project/issues/75676)
+
+## Introduction
+Resources are often used in HLSL, with various resource element types (RETs).
+For example:
+```
+RWBuffer<float> rwbuf: register(u0);
+```
+In this code, the RET is `float`, and the resource type is `RWBuffer`. The
+resource type is not a `RawBuffer` variant, and so there is a distinct set
+of rules that define valid RETs for this resource type.
+
+RETs for non-`RawBuffer` variants may include basic types (ints and uints of sizes 16

```

Comment:
Maybe a list of resource types and what category they fall into would be helpful here?

I'm wondering if DXIL.rst needs to be updated to not refer to "RawBuffer" as an HLSL resource type.  Maybe it means ByteAddressBuffer?

DXIL.rst talks about "typed buffers".  Is that the right term for a "not raw buffer"?

Reply:
I think the full list would be way too long and would overpower this spec. There's a lot of resource types. 
It's strange, I think there are some types in that file that refer to a set of actual spellable types, but of course, RawBuffer isn't actually spellable. I don't think it's a typo.
Perhaps typed buffers is the right term, I'll double check.

Reply:
Given a random buffer type, how do I determine if it is raw or typed?  I really think just having a list of the HLSL data types and whether they're raw or typed would help here.  I don't even think it is long, but obfuscating it in prose doesn't help.

* `[Append|Consume|RW]StructuredBuffer` - raw
* `[RW]Buffer` - typed
* `[RW]ByteAddressBuffer` - raw?
* `[Feedback]Texture*` - typed (I assume Feedback textures are typed)